CONTRIBUTING.md

```diff
@@ -1,68 +1,80 @@
+
+If there is not an open issue for what you want to submit, prefer opening one for discussion before working on a PR. You
+can work on any issue that doesn't have an open PR linked to it or a maintainer assigned to it. These show up in the
+sidebar. No need to ask if you can work on an issue that interests you.
 
 ## Branches
-When starting development on this project, please make sure to create a new branch off the `dev` branch. This helps to keep the `master` branch stable and free of any development work that may not be complete or fully tested.
+
+Make sure to create a branch off your master branch, and then submit a PR to our master branch.
 
 ## Project Structure
+
 ```
 .
 ├── app                      # Backend code (FastAPI - Python)
-│   └── dashboard            # Frontend code (React - Typescript)
-├── cli                      # CLI code (Typer - Python)
-└── xray_api                 # Client of Xray's gRPC API
+├── dashboard                # Frontend code (React - Typescript)
+└── cli                      # CLI code (Typer - Python)
 ```
 
 ## Backend
-Backend is built using FastAPI and uses SQLAlchemy as the ORM for database operations. All Pydantic models can be found in the `app/models` directory, while all database-related operations and models are in the `app/db` directory. The migration scripts for the database (Alembic) can be found in the `app/db/migrations` directory.
+
+Backend is built using FastAPI and uses SQLAlchemy as the ORM for database operations. Pydantic models can be found
+in the `app/models` directory, while all database-related operations and models are in the `app/db` directory. The
+migration scripts for the database (Alembic) can be found in the `app/db/migrations` directory.
 
 ### Python Code Formatting
-To maintain consistency in the codebase, we require all code to be formatted using 
+
+To maintain consistency in the codebase, we require all code to be formatted using
+
 ```bash
-autopep8 <file> --max-line-length 120
+black .
 ```
 
 ## Frontend
-Frontend is pre-built and served by FastAPI from the `app/dashboard/build` directory. To rebuild the frontend, first make sure you have the necessary dependencies installed by running `npm install` in the `app/dashboard` directory. Then, simply remove the `app/dashboard/build` directory and run the Python code again, and it will rebuild the frontend automatically.
 
-### Components Library
-Frontend uses `Chakra-UI` as the component library, so please adhere to the Chakra-UI approach when contributing. Strive to create components that are cohesive and serve a single purpose. Keep in mind that readability and maintainability are more important than brevity, so prioritize those factors when writing your code.
+To build the frontend you'll need the following dependencies:
+
+- npm
```

Comment:
We should get ride of npm, it is completely unnecessary to require developers and users to use two package manager that are alternative to each other.